hypervisor/src/arch/aarch64/gic.rs

```diff
@@ -36,6 +39,41 @@ pub struct VgicConfig {
     pub nr_irqs: u32,
 }
 
+#[derive(Clone, Serialize)]
+pub enum GicState {
+    #[cfg(feature = "kvm")]
+    Kvm(crate::kvm::aarch64::gic::Gicv3ItsState),
+}
+
+impl<'de> Deserialize<'de> for GicState {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value: serde_json::Value = Deserialize::deserialize(deserializer)?;
+
+        #[cfg(feature = "kvm")]
+        if let Ok(gicv3_its_state) =
+            crate::kvm::aarch64::gic::Gicv3ItsState::deserialize(value.clone())
+        {
+            return Ok(GicState::Kvm(gicv3_its_state));
+        }
+
+        #[cfg(feature = "kvm")]
```

Comment:
Is it possible to use a type alias to e.g. `GicStateReal` and then `#[derive(Deserialize)]` on that and then alias the results together (might need to `std::mem::transmute` but if they have identical structures that's fine) so you don't need to manually deserialize - this would mean we could keep this code around for much longer.

Reply:
I mean that type alias will break, if compile with just `cargo build --no-default-features --features="mshv"`, that's why I was not in an opinion of maintaining this code in long term. Or we just move from these feature flags and make mshv and kvm built-in by default.

Reply:
@rbradford How about the new revision?

Reply:
I think this is good - now you don't have to manually implement. I know you used transmute as I suggested here - we could also do it safely with an `impl From`

Reply:
Done